### PR TITLE
feat: add saveAs option

### PR DESCRIPTION
**Pull request by ranjith-sg**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -812,6 +861,35 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
                 }
             }
         });
+        if ($meta.saveAs) {
+            var fileDir = path.dirname($meta.saveAs);
+            return new Promise((resolve, reject) => {
+                fs.access(fileDir, (err) => {
```
Comment (kalinkrustev) [inline index.js]: I think it is better to directly try to create the dir and proceed normally in case of error.code==='EEXIST'
Check notes here: https://nodejs.org/api/fs.html#fs_fs_access_path_mode_callback
The notes are about file, but should be the same for folders.

File: index.js [hunk 2]
```diff
@@ -812,6 +861,35 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
...
+                            return resolve();
+                        }
+                        reject(e);
+                    });
+                });
+            }).then(function(resolve, reject) {
+                request.stream = true;
+                var ws = fs.createWriteStream(fileName);
+                request.execute(name);
+                transform(request).pipe(ws);
```
Comment (kalinkrustev) [inline index.js]: I think you need to pipe before execute. At least their example is doing so: https://github.com/patriksimek/node-mssql#pipe-stream

File: index.js (hunk 1 above)
Comment (ranjith-sg) [reply index.js]: Please check it now

File: index.js (hunk 2 above)
Comment (ranjith-sg) [reply index.js]: Please check it now

File: index.js [hunk 3]
```diff
@@ -812,6 +861,32 @@ SqlPort.prototype.callSP = function(name, params, flatten, fileName) {
...
+                request.execute(name);
+                return new Promise(function(resolve, reject) {
+                    ws.on('finish', function() {
+                        return resolve({fileName: $meta.saveAs});
+                    });
+                    ws.on('error', function(err) {
+                        return reject(err);
+                    });
+                });
+            }).catch(function(err) {
```
Comment (kalinkrustev) [inline index.js]: Why did you add this catch? This will result in promise being resolved with the error, which I doubt to be expected.

Comment (ranjith-sg) [reply index.js]: Sorry, Yes. It should be removed and may be handle it on consumer end if needed. Now I have removed the catch section please check it.

**Comment by kalinkrustev** [comment]
OK, now all is good. As we are now in code freeze, I will postpone merging this. Meanwhile I was working on a new major version (in major/ut6 branch), where I have added your code.
